src/ucs/datastruct/bitmap.h

```diff
@@ -449,6 +455,36 @@ static UCS_F_ALWAYS_INLINE bool _ucs_bitmap_is_zero(const void *bitmap, size_t n
 }
 
 
+/**
+ * Find the index of the first bit set to 1 in a given bitmap, starting from
+ * a particular index (excluding it)
+ *
+ * @param bitmap_words Look for the first bit in the words of this bitmap
+ * @param num_words    Number of words in the bitmsp
+ * @param start_index  The first bit to look from
+ */
+static UCS_F_ALWAYS_INLINE int
+ucs_bitmap_ffs(const ucs_bitmap_word_t *bitmap_words, size_t num_words,
+               size_t start_index)
+{
+    size_t word_index = start_index / UCS_BITMAP_BITS_IN_WORD;
+    size_t mask       = ~UCS_MASK(start_index % UCS_BITMAP_BITS_IN_WORD);
+
+    while (word_index < num_words) {
+        if (bitmap_words[word_index] & mask) {
+            return _UCS_BITMAP_BIT_INDEX(ucs_ffs64(bitmap_words[word_index] &
+                                                   mask),
```

Comment:
minor: can add helper var to store result of ucs_ffs64

Reply:
Done